OpenSim/Common/TableSource.h

```diff
@@ -0,0 +1,214 @@
+/* -------------------------------------------------------------------------- *
+ *                            OpenSim:  TableSource.h                         *
+ * -------------------------------------------------------------------------- *
+ * The OpenSim API is a toolkit for musculoskeletal modeling and simulation.  *
+ * See http://opensim.stanford.edu and the NOTICE file for more information.  *
+ * OpenSim is developed at Stanford University and supported by the US        *
+ * National Institutes of Health (U54 GM072970, R24 HD065690) and by DARPA    *
+ * through the Warrior Web program.                                           *
+ *                                                                            *
+ * Copyright (c) 2005-2015 Stanford University and the Authors                *
+ * Authors:                                                                   *
+ *                                                                            *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may    *
+ * not use this file except in compliance with the License. You may obtain a  *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.         *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing, software        *
+ * distributed under the License is distributed on an "AS IS" BASIS,          *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ * -------------------------------------------------------------------------- */
+
+#ifndef OPENSIM_TABLE_SOURCE_H_
+#define OPENSIM_TABLE_SOURCE_H_
+
+#include "TimeSeriesTable.h"
+#include "Component.h"
+
+namespace OpenSim {
+
+class TimeOutOfRange : public InvalidTimestamp {
+public:
+    TimeOutOfRange(const std::string& file,
+                   size_t line,
+                   const std::string& func,
+                   double timestamp,
+                   double minTimestamp,
+                   double maxTimestamp) :
+        InvalidTimestamp(file, line, func) {
+        std::string msg = "min = " + std::to_string(minTimestamp);
+        msg += " max = " + std::to_string(maxTimestamp);
+        msg += " timestamp = " + std::to_string(timestamp);
+
+        addMessage(msg);
+    }
+}; // class TimeOutOfRange
+
+
+/** Component representing a source of data from a TimeSeriesTable_.
+
+This Component has two outputs:
+- A list output wih one channel per column of the TimeSeriesTable_. 
```

Comment:
I found these descriptions unhelpful. What is a "list output"? What do you mean by "channel"? I think the descriptions of the outputs should be more user-friendly. Also, I think they should appear in the doxygen for the methods themselves. Here, I would describe (at a high level) what the outputs can return (e.g., "This component has outputs for retrieving all or a specified subset of the columns in a TimeSeriesTable_ at a specified time, where the values are interpolated in time if necessary."). :snowman: